setup/gcp/installer/forseti_server_installer.py

```diff
@@ -0,0 +1,339 @@
+
+            instance_name = '{}-vm'.format(deployment_name)
+            self.wait_until_vm_initialized(instance_name)
+
+            # Create firewall rule to block out all the ingress traffic
+            create_firewall_rule(
+                self.format_firewall_rule_name('forseti-server-deny-all'),
+                [self.gcp_service_account],
+                FirewallRuleAction.DENY,
+                ['icmp', 'udp', 'tcp'],
+                FirewallRuleDirection.INGRESS,
+                1)
+
+            # Create firewall rule to allow only port tcp:50051
+            create_firewall_rule(
+                self.format_firewall_rule_name('forseti-server-allow-grpc'),
+                [self.gcp_service_account],
+                FirewallRuleAction.ALLOW,
+                ['tcp:50051'],
+                FirewallRuleDirection.INGRESS,
+                0)
+
+        return success, deployment_name
+
+    def format_firewall_rule_name(self, rule_name):
+        """Format firewall rule name.
+
+        Args:
+            rule_name (str): Name of the firewall rule
+
+        Returns:
+            str: Firewall rule name
+        """
+        return '{}-{}'.format(rule_name, self.config.datetimestamp)
+
+    def should_setup_explain(self):
+        """Ask user if they want to configure setup for Explain."""
+        print_banner('Enable IAM Explain')
+        print(MESSAGE_SETUP_IAM_EXPLAIN)
+        choice = None
+        if not self.config.advanced_mode:
+            choice = 'y'
+            print('Automatically enabling IAM Explain (basic setup mode).')
+
+        while not choice:
+            choice = raw_input(QUESTION_ENABLE_IAM_EXPLAIN).strip()
+        self.setup_explain = choice == 'y'
+
+    def inform_access_on_target(self):
+        """Inform user that they need IAM access to grant Forseti access."""
+        print_banner('Current IAM access')
+        choice = None
+
+        if not self.config.advanced_mode:
+            choice = 'y'
+
+        while choice != 'y' and choice != 'n':
+            choice = raw_input(QUESTION_ACCESS_TO_GRANT_ROLES.format(
+                self.resource_root_id)).strip().lower()
+
+        if choice == 'y':
+            self.user_can_grant_roles = True
+            print('Will attempt to grant roles on the target %s.' %
+                  self.resource_root_id)
+        else:
+            self.user_can_grant_roles = False
+            print('Will NOT attempt to grant roles on the target %s.' %
+                  self.resource_root_id)
+
+    def get_deployment_values(self):
+        """Get deployment values
+
+        Returns:
+            dict: A dictionary of values needed to generate
+                the forseti deployment template
+        """
+        bucket_name = self.generate_bucket_name()
+        return {
+            'CLOUDSQL_REGION': self.config.cloudsql_region,
+            'CLOUDSQL_INSTANCE_NAME': self.config.cloudsql_instance,
+            'SCANNER_BUCKET': bucket_name[len('gs://'):],
```

Comment:
Nit, dropping this here for lack of a better place. This is no longer just scanner configs, I would rename this globally to FORSETI_CONFIGURATION_BUCKET or something like that.